src/cache.ts

```diff
@@ -8,17 +8,25 @@ interface DecodedToken {
   user: any;
 }
 
-interface CacheEntry extends CacheKeyData {
+interface CacheEntry {
```

Comment:
Curious as to why we we went with flattening this class.

Reply:
I don't see the two types as being linked in a way that there needs to be inheritance here. To me it looks like the inheritance exists just because they have common properties, but I'm not sure that's a good enough case. It might make it harder to change later. There are lots of examples of this throughout the codebase I'd like to clean up, but I agree it's a case of early optimization.